From d862909c9cda9127c75d7a181fa748ba0a2a017d Mon Sep 17 00:00:00 2001
From: Tirumalesh Chalamarla <tchalamarla@caviumnetworks.com>
Date: Tue, 9 Oct 2018 00:40:51 +0300
Subject: [PATCH 0251/1051] octeontx-fpa: fix remove driver

Release resources when fpa vf driver is unbind

Signed-off-by: Tirumalesh Chalamarla <tchalamarla@caviumnetworks.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
index 6fdd0d35d658..116dc2a16cd6 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
@@ -381,6 +381,7 @@ static void fpavf_irq_free(struct fpavf *fpa)
 
 	free_irq(fpa->msix_entries[0].vector, fpa);
 	pci_disable_msix(fpa->pdev);
+	devm_kfree(&fpa->pdev->dev, fpa->msix_entries);
 }
 
 static int fpavf_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
@@ -461,6 +462,11 @@ static void fpavf_remove(struct pci_dev *pdev)
 	spin_unlock(&octeontx_fpavf_devices_lock);
 
 	fpavf_irq_free(fpa);
+	pcim_iounmap(pdev, fpa->reg_base);
+	pci_disable_device(pdev);
+	pci_release_regions(pdev);
+
+	devm_kfree(&pdev->dev, fpa);
 }
 
 /* devices supported */
-- 
2.17.1

